### PR TITLE
Surf

### PR DESCRIPTION
**Pull request by Artem-Zhdanov**
Now DeBot works with Surf

### CODE REVIEW

File: simple-todo-app/README.md [hunk 1]
```diff
@@ -5,57 +5,70 @@ This is an example of a simple TODO application. The application consists of two
...
+![](net.ton.dev.svg)
+
+### main.ton.dev.
+DeBot address: 0:73a7ba235ac26029574f0e053b3f25ba4d536b8ba2c8dd5d10fb266c9035bc36
+
+Open a link: https://uri.ton.surf/debot?address=0%3A73a7ba235ac26029574f0e053b3f25ba4d536b8ba2c8dd5d10fb266c9035bc36 
+
+![](main.ton.dev.svg)
+
+-   On the first launch DeBot deploys TODO contract with initial balance = 0.2 Ruby, so you have a Surf wallet with positive balance.
```
Comment (Keshoid) [inline simple-todo-app/README.md:26]: Why ruby? Use `ton tokens`.

Comment (Keshoid) [reply simple-todo-app/README.md:26]: `so you have a Surf wallet`. -> `so you need to have a Surf wallet ...`

File: simple-todo-app/todoDebot.sol [hunk 2]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
+            AddressInput.get(tvm.functionId(creditAccount),"Select a wallet for payment, we'll ask you to sign two transactions");
+
+        } else  if (acc_type == 0) { // acc is uninitialized
+            Terminal.print(0, format(
+                "Deploying new contract, if an error occured, check if your TODO contract has enough tokens on its balance"
+            ));
+            deploy();
+
+        } else if (acc_type == 2) {  // acc is frozen
+            Terminal.print(0, format("Account {} is frozen, send some tokens", m_address));
```
Comment (Keshoid) [inline simple-todo-app/todoDebot.sol:109]: Bad instructions. Just say that you cannot continue.

File: simple-todo-app/todoDebot.sol [hunk 3]
```diff
@@ -206,10 +319,21 @@ contract TodoDebot is Debot {
...
-        (name, semver) = ("TODO DeBot", _version(0,1,0));
+        (name, semver) = ("TODO DeBot", _version(0,2,0));
     }
 
     function _version(uint24 major, uint24 minor, uint24 fix) private pure inline returns (uint24) {
         return (major << 16) | (minor << 8) | (fix);
     }
+
+	// Function that changes the code of current contract.
+	function setCode(TvmCell newcode) public view  {
```
Comment (Keshoid) [inline simple-todo-app/todoDebot.sol:330]: You can inherit from `Upgradable` contract.

File: simple-todo-app/todoDebot.sol [hunk 4]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
 
-    function enterTodoAddr(address value) public {
-        m_address = value;
-        _getStat(tvm.functionId(setStat));
-    }
-
     function _menu() private {
         string sep = '----------------------------------------';
-        Terminal.print(0, sep);
+        // Terminal.print(0, sep);
```
Comment (Keshoid) [inline simple-todo-app/todoDebot.sol:178]: Remove if not used

File: simple-todo-app/todoDebot.sol [hunk 5]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
 
     function start() public override {
-        AddressInput.get(tvm.functionId(enterTodoAddr),"Enter your TODO contract address");
+        Terminal.input(tvm.functionId(savePublicKey),"Please enter your public key",false);
+    }
+
+    function savePublicKey(string value) public {
+        uint res;
+        bool status;
+        (res, status) = stoi("0x"+value);
```
Comment (Keshoid) [inline simple-todo-app/todoDebot.sol:78]: Write in one line:
(uint res, bool status) = stoi(...);

**Review by Keshoid — CHANGES_REQUESTED** [review]
See comments

File: simple-todo-app/README.md [hunk 6]
```diff
@@ -5,57 +5,70 @@ This is an example of a simple TODO application. The application consists of two
...
+This DeBot is already deployed on blockchain
 
- Install tondev globally
- ```
- $ npm i tondev -g
- ```
+### net.ton.dev
+DeBot address: 0:ba06ea5c648ff6149e75a0a589becd827a2d959e42a34eb5e6ee29cb080bc552
+
+Open a link: https://uri.ton.surf/debot?address=0%3Aba06ea5c648ff6149e75a0a589becd827a2d959e42a34eb5e6ee29cb080bc552&net=devnet
```
Comment (a-zorina) [inline simple-todo-app/README.md:15]: here and below: Open a link -> Open the link

File: simple-todo-app/todoDebot.sol [hunk 7]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
+        if (status) {
+            m_masterPubKey = res;
+
+            Terminal.print(0, "Checking if you already have TODO list ...");
```
Comment (a-zorina) [inline simple-todo-app/todoDebot.sol:82]: Checking if you already have a TODO list

File: simple-todo-app/todoDebot.sol [hunk 8]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
+        }
+    }
+
+
+    function checkStatus(int8 acc_type) public {
+        if (acc_type == 1) { // acc is active and  contract is already deployed 
+            _getStat(tvm.functionId(setStat));
+
+        } else if (acc_type == -1)  { // acc is inactive
+            Terminal.print(0, "You don't have TODO list yet, so new contract with an initial balance of 0.2 will be deployed");
```
Comment (a-zorina) [inline simple-todo-app/todoDebot.sol:99]: You don't have a TODO list yet, so a new contract with an initial balance of 0.2 tokens will be deployed

File: simple-todo-app/todoDebot.sol [hunk 9]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
+            AddressInput.get(tvm.functionId(creditAccount),"Select a wallet for payment, we'll ask you to sign two transactions");
```
Comment (a-zorina) [inline simple-todo-app/todoDebot.sol:100]: Select a wallet for payment. We will ask you to sign two transactions.

File: simple-todo-app/todoDebot.sol [hunk 10]
```diff
@@ -31,37 +42,140 @@ interface ITodo {
...
+
+        } else  if (acc_type == 0) { // acc is uninitialized
+            Terminal.print(0, format(
+                "Deploying new contract, if an error occured, check if your TODO contract has enough tokens on its balance"
```
Comment (a-zorina) [inline simple-todo-app/todoDebot.sol:104]: Deploying new contract. If an error occurs, check if your TODO contract has enough tokens on its balance

File: simple-todo-app/todoDebot.sol (hunk 2 above)
Comment (a-zorina) [reply simple-todo-app/todoDebot.sol:109]: I suggest "Cannot continue: account {} is frozen"

**Comment by Artem-Zhdanov** [comment]
All comment were accepted. I believe it's time to merge

**Review by Keshoid — APPROVED** [review]
LGTM

**Pull request merged**